ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -174,17 +179,29 @@ private[ember] object Parser {
         var httpVersion: HttpVersion = null
         val upperBound = Math.min(message.size - 1, maxHeaderSize)
 
+        def matchesString(start: Int, end: Int, expected: String): Boolean =
+          (end - start == expected.length) && {
+            var ix = 0
+            var isSame = true
+            while (start + ix < end && isSame) {
+              isSame = message(start + ix).toChar == expected.charAt(ix)
+              ix += 1
+            }
+            isSame
+          }
+
         var start = 0
         while (!complete && idx <= upperBound) {
           val value = message(idx)
           (state: @switch) match {
             case 0 =>
               if (value == space) {
-                Method.fromString(new String(message, start, idx - start)) match {
-                  case Left(e) =>
-                    throwable = e
+                frequentMethods.find(m => matchesString(start, idx, m.name)) match {
```

Comment:
I feel like linear search would be slower than getting value by key in a hash map. Wouldn't that brings some regression of speed in the parsing?

Reply:
That is what I started to wonder. That is why I added that `frequentMethods` list, altered in favour of the frequent four methods, GET, POST, PUT, and DELETE. Not sure if allocation is more expensive than building the string and computing its hash.